internal/transport/http_util.go

```diff
@@ -403,6 +429,26 @@ func newFramer(conn net.Conn, writeBufferSize, readBufferSize int, maxHeaderList
 	return f
 }
 
+func getWriteBufferPool(writeBufferSize int, shareWriteBuffer bool) *sync.Pool {
+	if !shareWriteBuffer {
+		return nil
+	}
+	writeBufferMutex.Lock()
+	defer writeBufferMutex.Unlock()
+	size := writeBufferSize * 2
+	pool, ok := writeBufferPoolMap[size]
+	if !ok {
+		pool = &sync.Pool{
+			New: func() interface{} {
+				b := make([]byte, size)
+				return &b
+			},
+		}
+		writeBufferPoolMap[size] = pool
+	}
+	return pool
```

Comment:
Restructure like this:

```golang
	pool, ok := writeBufferPoolMap[size]
	if ok {
		return pool
	}
	writeBufferPoolMap[size] = &sync.Pool{
		New: func() interface{} {
			b := make([]byte, size)
			return &b
		},
	}
	return writeBufferPoolMap[size]
```

I'm also OK with assigning the newly created `sync.Pool` to the local var `pool` and writing that to the map and returning it (as you currently have). The crux of this comment though is https://google.github.io/styleguide/go/decisions#indent-error-flow.

Reply:
done